### ISSUE TITLE
"Hide QuPath" on macOS opens "About" window instead of hiding window

### ISSUE THREAD

**Issue by tbedau** [qupath/qupath#1436]
**Describe the bug**
When trying to hide QuPath via the menu bar or by pressing cmd + h, the "About" window is opened instead of QuPath being hidden.

**To Reproduce**
1. Click on app name ("QuPath-0.5.0-arm64") in menu bar.
2. Click on "Hide QuPath-0.5.0-arm64" (alternatively: press cmd + h).
4. See error.

**Expected behavior**
QuPath window being hidden.

**Screenshots**
<img width="553" alt="qupath-bug" src="https://github.com/qupath/qupath/assets/69714383/7e5be5b2-1bd6-408f-8f56-46a1ff4c3f67">

**Desktop (please complete the following information):**
 - OS: macOS 14.1.2
 - QuPath Version: 0.5.0-arm64

**Comment by petebankhead** [comment]
I can replicate the problem, but I can't explain it and have no idea where it could be influenced in the QuPath code :/
I'd think it's a JavaFX bug, except I see it didn't occur in QuPath v0.4.x. But since we don't populate that system menu explicitly in QuPath, I don't know how we can change its behavior.

**Comment by tbedau** [comment]
That is very strange indeed. Unfortunately I don't have the needed knowledge to dive into this, but I guess a lot of heavy cmd+h users like me would be extremely grateful if someone found a solution!

**Comment by Rylern** [comment]
This bug is not specific to QuPath (see [here](https://stackoverflow.com/questions/55068825/handling-commandh-programmatically-in-mac-os)) so I also think it's a JavaFX bug.

A workaround would be to disable this "About" window and handle explicitly the "cmd+h" combination with something like:

```java
// Disable about window
Desktop.getDesktop().setAboutHandler(e -> {});

// Handle cmd + h
scene.getAccelerators().put(
  new KeyCodeCombination(KeyCode.H, KeyCombination.META_DOWN), () -> {
    stage.setIconified(true);
  }
);
```

inserted in the [constructor of QuPathGui](https://github.com/qupath/qupath/blob/13bdeed047b4d05f35f47308b36b48c0f2bb3a24/qupath-gui-fx/src/main/java/qupath/lib/gui/QuPathGUI.java#L256C10-L256C19).

However, this would still make impossible to hide QuPath via the menu bar.

**Comment by petebankhead** [comment]
I've just built v0.6.0-SNAPSHOT from source and this problem seems to have gone anyway... I guess through a Java/JavaFX dependency update.

So I think we can close this since it'll be fixed in the next release (even though we didn't intentionally change anything to fix it)?

**Comment by petebankhead** [comment]
Closing this as the problem should be fixed in v0.6.0 (somehow).
